src/sweeps/params.py

```diff
@@ -366,41 +366,29 @@ def to_config(self) -> Dict:
         """Convert a HyperParameterSet to a SweepRun config."""
         return dict([param._to_config() for param in self])
 
-    def convert_runs_to_normalized_vector(self, runs: List[SweepRun]) -> ArrayLike:
-        """Converts a list of SweepRuns to an array of normalized parameter vectors.
-
-        Args:
-            runs: List of runs to convert.
-
-        Returns:
-            A 2d array of normalized parameter vectors.
-        """
-
-        runs_params = [run.config for run in runs]
-        X = np.zeros([len(self.searchable_params), len(runs)])
-
-        for key, bayes_opt_index in self.param_names_to_index.items():
-            param = self.param_names_to_param[key]
-            row = np.array(
-                [
-                    (
-                        param.value_to_int(config[key]["value"])
-                        if param.type == HyperParameter.CATEGORICAL
-                        else config[key]["value"]
-                    )
-                    if key in config
-                    # filter out incorrectly specified runs
-                    else np.nan
-                    for config in runs_params
-                ]
-            )
-
-            X_row = param.cdf(row)
-            # only use values where input wasn't nan
-            non_nan = ~np.isnan(row)
-            X[bayes_opt_index, non_nan] = X_row[non_nan]
-
-        return np.transpose(X)
+    def normalize_runs_as_array(self, runs: List[SweepRun]) -> np.ndarray:
+        """Normalize a list of SweepRuns to an ndarray of parameter vectors."""
+        normalized_runs: np.ndarray = np.zeros([len(self.searchable_params), len(runs)])
+        for param_name, bayes_opt_index in self.param_names_to_index.items():
+            _param: HyperParameter = self.param_names_to_param[param_name]
+            row: np.ndarray = np.zeros(len(runs))  # default to 0
+            for i, run in enumerate(runs):
+                if param_name in run.config:
+                    _val = run.config[param_name]["value"]
+                    if _param.type == HyperParameter.CATEGORICAL:
+                        row[i] = _param.value_to_idx(_val)
+                    else:
+                        row[i] = _val
+                else:
+                    logging.warning(f"Run does not contain parameter {param_name}")
+            if not np.all(np.isfinite(row)):
+                logging.warning(f"Found non-finite value in normalized run row {row}")
+            # Convert row to CDF, filter out NaNs
+            non_nan_indices = ~np.isnan(row)
+            normalized_runs[bayes_opt_index, non_nan_indices] = _param.cdf(row)[
+                non_nan_indices
+            ]
```

Comment:
i think row here can contain `nan`/`None`, let's pass `row[non_nan_indices]` to `_param.cdf` (to ensure we never get the original exception that triggered things here) instead of indexing the result of a potentially unsafe call to `_param.cdf`
```suggestion
            normalized_runs[bayes_opt_index, non_nan_indices] = _param.cdf(row[
                non_nan_indices
            ])
```